platform-hub-api/spec/controllers/kubernetes/groups_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe Kubernetes::GroupsController, type: :controller do
+
+  describe 'GET #index' do
+
+    it_behaves_like 'unauthenticated not allowed' do
+      before do
+        get :privileged
+      end
+    end
+
+    it_behaves_like 'authenticated' do
+
+      it_behaves_like 'not an admin so forbidden'  do
+        before do
+          get :privileged
+        end
+      end
+
+      it_behaves_like 'an admin' do
+        before do
+          create(:kubernetes_groups_hash_record)
+        end
+
+        it 'returns the list of cluster-wide privileged kubernetes groups' do
+          get :privileged
+          expect(response).to be_success
+
+          group = json_response.first
+
+          expect(group['id']).to eq 'acp:all:user:cw-privileged'
```

Comment:
Right now this is assuming a value from the factory, which would break if we changed the factory. (In theory, default values provided by a factory are for fields you don't care about). Maybe set the groups explicitly above and check for an explicit value that's set? This way you can also add a non-privileged group too and check that isn't returned. (I.e. need to also check number of responses to make sure it's been filtered appropriately).